lib/improver/nbhood/recursive_filter.py

```diff
@@ -337,13 +339,13 @@ def process(self, cube, alphas_x=None, alphas_y=None, mask_cube=None):
                     output, mask_cube))
             mask = mask.data.squeeze()
 
-            padded_cube = SquareNeighbourhood().pad_cube_with_halo(
-                output, self.edge_width, self.edge_width)
+            padded_cube = pad_cube_with_halo(
+                output, 2*self.edge_width, 2*self.edge_width)
```

Comment:
Recursive filtering is not the same use case as square neighbourhooding - I'm not sure what the actual extent of the recursion is.  So I can try this, but I'm not convinced it will give me the same final answer.

Reply:
Maybe a bit clearer - I've changed what the square neighbourhood plugin is actually doing internally, but this does not impact the final result (all the impacts are in the halo itself which gets chopped off again before returning the neighbourhooded cube).  However the recursive filter pulls in information from longer ranges in a different (recursive...) way.  So I didn't want to change anything about the intermediates as I think there's a good chance it will influence the final result.

Do either of those answers make sense?  Let me know if you want me to try this, otherwise I'll leave it to be safe.